src/firefly/js/templates/lightcurve/LcPeriod.jsx

```diff
@@ -0,0 +1,1035 @@
+    } else {
+        pd = periodRange.min;
+    }
+    if (tzero) {
+        if (!tzero.valid) return {};      // invalid period
+        tz = parseFloat(tzero.value);
+    } else {
+        tz = periodRange.tzero;
+    }
+
+    var phaseV = data.map((d) => {
+        return parseFloat(getPhase(d[tIdx], tz, pd));
+    });
+
+    var minPhase = Math.min(...phaseV);
+    var maxPhase = Math.max(...phaseV) + 1.0;
+
+    var xy = data.reduce((prev, d, idx) => {
+        var phase1, phase2, fluxV;
+
+        fluxV = parseFloat(d[fIdx]);
+        phase1 = phaseV[idx];
+        phase2 = phase1 + 1.0;
+        prev.push([phase1, fluxV], [phase2, fluxV]);
+
+        return prev;
+    }, []);
+
+    return Object.assign({data: xy}, {minPhase, maxPhase, period: pd.toFixed(DEC_PHASE), flux: fc});
+}
+
+
+/**
+ * @summary Phase folding finder component containing period finding parameters
+ */
+class LcPFOptionsBox extends Component {
+    constructor(props) {
+        super(props);
+
+        const fields = FieldGroupUtils.getGroupFields(pfinderkey);
+        const maxPeriod = fields && get(fields, ['periodMax', 'value']);
+        const minPeriod = fields && get(fields, ['periodMin', 'value']);
+        const period = fields && get(fields, ['period', 'value']);
+        this.state = {fields: FieldGroupUtils.getGroupFields(pfinderkey),
+                      minPeriod, maxPeriod, period};
+    }
+
+    componentWillUnmount() {
+        this.iAmMounted= false;
+        if (this.unbinder) this.unbinder();
+    }
+
+
+    componentDidMount() {
+        this.iAmMounted= true;
+        this.unbinder= FieldGroupUtils.bindToStore(pfinderkey, (fields) => {
+            if (this && this.iAmMounted && fields!==this.state.fields) {
+
+                const maxPeriod = fields && get(fields, ['periodMax', 'valid']) && get(fields, ['periodMax', 'value']);
+                const minPeriod = fields && get(fields, ['periodMin', 'valid']) && get(fields, ['periodMin', 'value']);
+                const period =  fields && get(fields, ['period', 'valid']) && get(fields, ['period', 'value']);
+                var   newState = {fields};
+
+                if (maxPeriod) {
+                    newState = Object.assign({}, newState, {maxPeriod});
+                }
+                if (minPeriod) {
+                    newState = Object.assign({}, newState, {minPeriod});
+                }
+                if (period) {
+                    newState = Object.assign({}, newState, {period})
+                }
+
+                this.setState(newState);
+            }
+        });
+    }
+
+    render() {
+        var {minPeriod, maxPeriod, period} = this.state;   // only pass valid minPeriod & maxPeriod defined previously
+
+        return (
+                <LcPFOptions minPeriod={minPeriod} maxPeriod={maxPeriod} period={period}/>
+        );
+
+    }
+}
+
+/**
+ * @summary light curve phase folding FieldGroup rendering
+ * @param period
+ * @param minPeriod
+ * @param maxPeriod
```

Comment:
this jsdoc is not valid.  there's only one param in this function.  there are also a few more places where validation fail (intelliJ 'red' indicators).